engine/src/test/java/io/camunda/zeebe/engine/processing/deployment/MultiPartitionDeploymentLifecycleTest.java

```diff
@@ -53,20 +54,32 @@ public void shouldTestLifecycle() {
     final var deploymentPartitionRecords =
         RecordingExporter.records().withPartitionId(1).limit(10).collect(Collectors.toList());
 
-    assertThat(deploymentPartitionRecords)
+    assertThat(deploymentPartitionRecords).hasSize(10);
+
+    assertThat(deploymentPartitionRecords.subList(0, 5))
         .extracting(Record::getIntent, Record::getRecordType)
-        .hasSize(10)
         .containsExactly(
             tuple(DeploymentIntent.CREATE, RecordType.COMMAND),
             tuple(ProcessIntent.CREATED, RecordType.EVENT),
             tuple(DeploymentIntent.CREATED, RecordType.EVENT),
             tuple(DeploymentDistributionIntent.DISTRIBUTING, RecordType.EVENT),
-            tuple(DeploymentDistributionIntent.DISTRIBUTING, RecordType.EVENT),
-            tuple(DeploymentDistributionIntent.COMPLETE, RecordType.COMMAND),
-            tuple(DeploymentDistributionIntent.COMPLETE, RecordType.COMMAND),
-            tuple(DeploymentDistributionIntent.COMPLETED, RecordType.EVENT),
-            tuple(DeploymentDistributionIntent.COMPLETED, RecordType.EVENT),
-            tuple(DeploymentIntent.FULLY_DISTRIBUTED, RecordType.EVENT));
+            tuple(DeploymentDistributionIntent.DISTRIBUTING, RecordType.EVENT));
+
+    assertThat(deploymentPartitionRecords.subList(5, 9))
+        .extracting(
+            Record::getIntent,
+            Record::getRecordType,
+            r -> ((DeploymentDistributionRecordValue) r.getValue()).getPartitionId())
+        .containsSubsequence(
+            tuple(DeploymentDistributionIntent.COMPLETE, RecordType.COMMAND, 2),
+            tuple(DeploymentDistributionIntent.COMPLETED, RecordType.EVENT, 2))
+        .containsSubsequence(
+            tuple(DeploymentDistributionIntent.COMPLETE, RecordType.COMMAND, 3),
+            tuple(DeploymentDistributionIntent.COMPLETED, RecordType.EVENT, 3));
+
+    assertThat(deploymentPartitionRecords.subList(9, deploymentPartitionRecords.size()))
```

Comment:
:x: Please assert 
`containsSequence( COMPLETED on partition 2, FULLY_DISTRIBUTED)`
`containsSequence( COMPLETED on partition 3, FULLY_DISTRIBUTED)`

Reply:
I don't think we can do this. The `FULLY_DISTRIBUTED` it not written once per partition. It is only written once on partition 1 when we have received all the `COMPLETE` commands from the other partitions, which is already asserted with the previous assertion.

Reply:
Couldn't this work?
```
        .containsSubsequence(
            tuple(DeploymentDistributionIntent.COMPLETE, RecordType.COMMAND, 2),
            tuple(DeploymentDistributionIntent.COMPLETED, RecordType.EVENT, 2)).
            tuple(DeploymentIntent.FULLY_DISTRIBUTED, RecordType.EVENT, 1));
        .containsSubsequence(
            tuple(DeploymentDistributionIntent.COMPLETE, RecordType.COMMAND, 3),
            tuple(DeploymentDistributionIntent.COMPLETED, RecordType.EVENT, 3)),
            tuple(DeploymentIntent.FULLY_DISTRIBUTED, RecordType.EVENT, 1));
```

Reply:
I have done something similar now 🙂